packages/vats/src/core/boot-psm.js

```diff
@@ -210,6 +209,8 @@ export const buildRootObject = (vatPowers, vatParameters) => {
       // to code to be evaluated after initial bootstrap.
       bridgeCoreEval(powersFor('bridgeCoreEval')),
     ]);
+    // xxx this doesn't ever resolve yet, due to a dropped promise in startPSM (datalock)
+    console.log('boot-psm fully resolved');
```

Comment:
I guess I wasn't clear; the datalock showed up when I tried to fix the fact that there's a dropped promise here. The code we shipped does not suffer from datalock, but it does use a dropped promise to escape from datalock.

The caller of this function doesn't get a promise for the completion of all the work that's started here, as one might expect. We rely on the kernel's ability to sense an empty promise queue at the end of a crank for things to work out right.

Does that log message really never show up?

Reply:
ah... now I remember: returning the promise resulted in things getting stuck. So yes, it makes sense that this log message doesn't show up. I'm not sure where the datalock is, though.